resultbrowser/src/js/resultbrowser.js

```diff
@@ -381,215 +512,218 @@
+                        .then((response) => {
+                            if (!response.ok) {
+                                throw new Error('Network response was not OK');
+                            }
+                            return response.text();
+                        })
+                        .then((data) => {
+                            const subMime = requestData.mimeType.substring(requestData.mimeType.indexOf('/') + 1),
                                 lang = /x?html/.test(subMime) ? 'html' : /xml/.test(subMime) ? 'xml' : /(javascript|json)$/.test(subMime) ? 'javascript' : /^css$/.test(subMime) ? 'css' : undefined,
-                                canBeautify = lang && (( /(ht|x)ml/.test(lang) && extras.beautify.html ) || ( 'javascript' === lang && extras.beautify.js) || ('css' === lang && extras.beautify.css));
+                                canBeautify = lang && ((/(ht|x)ml/.test(lang) && extras.beautify.html) || ('javascript' === lang && extras.beautify.js) || ('css' === lang && extras.beautify.css));
+
+                            disableBeautifyButton(!canBeautify);
+                            disableSelectAllButton(false);
+                            disableHighlightButton(!extras.highlight);
 
-                            $('#beautify').prop('disabled', !canBeautify);
-                            $('#selectResponseContent').prop('disabled', false);
-                            $('#highlightSyntax').prop('disabled', !extras.highlight);
-                            $requestText.text(data).removeClass().addClass(lang ? ('language-'+lang+' '+lang) : 'text').show();
+                            setText(requestText, data);
+                            requestText.classList.remove(...requestText.classList);
+                            requestText.classList.add(...(lang ? [`language-${lang}`, lang] : ['text']));
+                            show(requestText);
 
                             // feed the json viewer if the mime type indicates json-ish content (e.g. "application/json" or "application/<...>+json")
                             if (/^application\/(.+\+)?json$/.test(requestData.mimeType)) {
                                 jsonView.format(data, '#jsonViewerContent');
                             }
-                        },
-                        error : function(xhr,textStatus,errorThrown) {
-                            $requestText.hide();
-                            $('#errorMessage .filename').text(requestData.fileName);
-                            $('#errorMessage').show();
-                            centerErrorMessage();
-                        }
-                    });
+                        }).catch(() => {
+                            hide(requestText);
+                            getFirstElementByQuery("#errorMessage .filename").disabled = true;
+                            setText(getFirstElementByQuery("#errorMessage .filename"), requestData.fileName);
```

Comment:
Lookup to same element is done twice. Result of lookup may be stored local variable for caching.

Reply:
Done